src/graphql/mutations/password_reset.rs

```diff
@@ -0,0 +1,116 @@
+use async_graphql::{Context, Object};
+use sea_orm::{
+    ActiveValue, ColumnTrait, DatabaseConnection, EntityTrait, IntoActiveModel, QueryFilter,
+};
+use sendgrid::SGClient;
+use tracing::{event, Level};
+
+use crate::{
+    error::new_err,
+    schema::{self, password_reset_tokens, users},
+};
+
+#[derive(Default)]
+pub struct PasswordResetMutation;
+const RESET_URL_BASE: &str = "https://lumina.earth/reset?token=";
+#[Object(rename_fields = "snake_case", rename_args = "snake_case")]
+impl PasswordResetMutation {
+    #[graphql()]
+    pub async fn reset_password(
+        &self,
+        ctx: &Context<'_>,
+        email: String,
+    ) -> async_graphql::Result<&str> {
+        let db = ctx.data_unchecked::<DatabaseConnection>();
+        let s_g_client = ctx.data_unchecked::<SGClient>();
+
+        let user = users::Entity::find()
+            .filter(users::Column::Email.contains(&email))
+            .one(db)
+            .await?
+            .ok_or_else(|| new_err("USER_NOT_FOUND", &format!("User not found: {}", &email)))?;
+
+        let mut new_reset_token = password_reset_tokens::ActiveModel::default();
+        new_reset_token.id = ActiveValue::Set(uuid::Uuid::new_v4());
+        new_reset_token.user_id = ActiveValue::Set(user.id);
+
+        let token = match password_reset_tokens::Entity::insert(new_reset_token)
+            .exec_with_returning(db)
+            .await
+        {
+            Err(db_error) => {
+                event!(Level::ERROR, "{}", db_error);
```

Comment:
We don't need this as we already log errors in the `new_err` function

Reply:
And same for other instances where you are logging the error

Reply:
actually I will defend this, this provides more useful and readible information than the new_err function

we should log errors instead of sending them directly to the svelte client and just tell it that something failed

Reply:
They are logged inside of `new_err`

Reply:
the new_err logs do not show up in the std_out of the test and are less than helpful. I much prefer the tokio logs for debugging and it has actually helped me more than the new_err func